### ISSUE TITLE
Fix `property-layout-mappings` false negatives for property names in declaration values

### ISSUE THREAD

**Issue by regseb** [stylelint/stylelint#9219]
### What is the problem you're trying to solve?

- `stylelint.config.js`

  ```javascript
  export default {
    rules: {
      "property-layout-mappings": "flow-relative",
    },
  };
  ```

- `style.css`

  ```css
  div {
    right: 0;
    transition: right 200ms ease;

    &.active {
      right: -100px;
    }
  }
  ```

[`property-layout-mappings`](https://stylelint.io/user-guide/rules/property-layout-mappings) only reports the use of the [`right`](https://developer.mozilla.org/Web/CSS/Reference/Properties/right) property, but it does not report the use of the value `right` within the [`transition`](https://developer.mozilla.org/Web/CSS/Reference/Properties/transition) property.

> 2:5-10 ERROR Disallowed physical property "right" [(property-layout-mappings)](https://stylelint.io/user-guide/rules/property-layout-mappings)
> 6:9-14 ERROR Disallowed physical property "right" [(property-layout-mappings)](https://stylelint.io/user-guide/rules/property-layout-mappings)

If you use _automatically fix_, the `transition` will no longer work.

### What solution would you like to see?

`property-layout-mappings` should check physical properties (like `right`) in values of:

- [`transition`](https://developer.mozilla.org/Web/CSS/Reference/Properties/transition)
- [`transition-property`](https://developer.mozilla.org/Web/CSS/Reference/Properties/transition-property)
- [`will-change`](https://developer.mozilla.org/Web/CSS/Reference/Properties/will-change), I hadn't heard of it before, and it gives the following warning: _`will-change` is intended to be used as a last resort, in order to try to deal with existing performance problems. It should not be used to anticipate performance problems._
- and maybe others...

### Relative issues

- https://github.com/stylelint/stylelint/issues/5430
- https://github.com/stylelint/stylelint/issues/8429
- https://github.com/stylelint/stylelint/pull/8687

**Comment by jeddy3** [comment]
@regseb Thanks for the report. I've opened a PR to fix.
